### ISSUE TITLE
is this using CoNll dataset for training NER tag ? dose the code support ontonotes as well?

### ISSUE THREAD

**Issue by AjitAntony** [kamalkraj/BERT-NER#58]
CoNll dataset - can predict only  5 tags
ontonotes-  can predict  around 18 tags
will the code work fine if i just replace the dataset with ontonotes or are there any restriction in using ontonotes for training bert for predicting NER ?
im unable to find any ontonotes datasets do u have any ?

**Comment by kamalkraj** [comment]
https://github.com/kamalkraj/BERT-NER/issues/52
@ranjeetds has trained BERT ner using ontonotes 5.0 dataset.

**Comment by ranjeetds** [comment]
@ajbot2019 You cannot use ononotes directly - 
You can request dataset here - [OntoNotes 5.0](https://catalog.ldc.upenn.edu/LDC2013T19)
As conll is in BIO annotated format, you need to convert OntoNotes into BIO
For that you can use - [OntoNotes to BIO](https://github.com/yuchenlin/OntoNotes-5.0-NER-BIO) this repo

Once done replace train test and valid data files from here and it should work.

**Comment by AjitAntony** [comment]
> #52
> @ranjeetds has trained BERT ner using ontonotes 5.0 dataset.

@kamalkraj   Thank you .  could you direct  me on how to create custom named entity recognition using BERT ? 
1.how should i prepare the train /test/dev data  for custome NER.
Should my dataset be in this format

-DOCSTART- -X- -X- O

coffe - NNP I-NP DRINKS
pizza - NNP I-NP JUNKFood
 
After formatting the dataset with custom tags like DRINKS,JUNKFood will running the code work fine or should i modify anything in below part of code ?

def get_labels(self):
        return ["O", "B-MISC", "I-MISC",  "B-PER", "I-PER", "B-ORG", "I-ORG", "B-LOC", "I-LOC", "[CLS]", "[SEP]"]

**Comment by AjitAntony** [comment]
> @ajbot2019 You cannot use ononotes directly -
> You can request dataset here - [OntoNotes 5.0](https://catalog.ldc.upenn.edu/LDC2013T19)
> As conll is in BIO annotated format, you need to convert OntoNotes into BIO
> For that you can use - [OntoNotes to BIO](https://github.com/yuchenlin/OntoNotes-5.0-NER-BIO) this repo
> 
> Once done replace train test and valid data files from here and it should work.

@ranjeetds  Thank you . im unable to download the dataset due as only verified corporate user will be able to download it  .Is your BERT-NER repo having this code or the pre trained model ?

After  replacing  train test and valid data files  , should  we modify anything in below code section ? as i can see this section related to label in the code part .

def get_labels(self):
return ["O", "B-MISC", "I-MISC", "B-PER", "I-PER", "B-ORG", "I-ORG", "B-LOC", "I-LOC", "[CLS]", "[SEP]"]

**Comment by AjitAntony** [comment]
@ranjeetds @kamalkraj .can you help me on how to test after  pre training the bert for masked word prediction with custom corpus that was given in the bert git page .

Hi followed the instruction shared in Pre-training with BERT (https://github.com/google-research/bert#pre-training-with-bert)

i ran these 2 file as mentioned in the instruction ,
python create_pretraining_data.py
python run_pretraining.py

after running those 2 file i got below list of files in the folder output_dir=/tmp/pretraining_output

checkpoint
eval
eval_results.txt
events.out.tfevents.1575545953.26760d2fc979
graph.pbtxt
model.ckpt-0.data-00000-of-00001
model.ckpt-0.index
model.ckpt-0.meta
model.ckpt-20.data-00000-of-00001
model.ckpt-20.index
model.ckpt-20.meta

1.what should i do now to test and perform the prediction of masked words for the sentences in the sample_text.txt that was used as corpus to pretrain this masked LM .

i know how we can load the BertForMaskedLM to do the mask word prediction but how to do it for this pretrained model that i have got in the output_dir=/tmp/pretraining_output ?

when i tried to load the model from this output instead of using BertForMaskedLM.from_pretrained('bert-base-uncased') i got below errror

from transformers import BertForMaskedLM
BertNSP=BertForMaskedLM.from_pretrained('/content/drive/My Drive/bert_training/pretraining_output/')

when i import
Model name '/content/drive/My Drive/bert_training/pretraining_output/' was not found in model name list (bert-base-uncased, bert-large-uncased, bert-base-cased, bert-large-cased, bert-base-multilingual-uncased, bert-base-multilingual-cased, bert-base-chinese, bert-base-german-cased, bert-large-uncased-whole-word-masking, bert-large-cased-whole-word-masking, bert-large-uncased-whole-word-masking-finetuned-squad, bert-large-cased-whole-word-masking-finetuned-squad, bert-base-cased-finetuned-mrpc, bert-base-german-dbmdz-cased, bert-base-german-dbmdz-uncased).

We assumed '/content/drive/My Drive/bert_training/pretraining_output/config.json' was a path or url to a configuration file named config.json or a directory containing such a file but couldn't find any such file at this path or url.

**Comment by manelAffi** [comment]
> @ajbot2019 You cannot use ononotes directly -
> You can request dataset here - [OntoNotes 5.0](https://catalog.ldc.upenn.edu/LDC2013T19)
> As conll is in BIO annotated format, you need to convert OntoNotes into BIO
> For that you can use - [OntoNotes to BIO](https://github.com/yuchenlin/OntoNotes-5.0-NER-BIO) this repo
> 
> Once done replace train test and valid data files from here and it should work.
@kamalkraj @ranjeetds 
I followed all  steps but I got an empty onto.train.ner, onto.developmnt.ner and onto.test.ner .
is there any ideas what should be the problem?

**Comment by manelAffi** [comment]
> @ajbot2019 You cannot use ononotes directly -
You can request dataset here - OntoNotes 5.0
As conll is in BIO annotated format, you need to convert OntoNotes into BIO
For that you can use - OntoNotes to BIO this repo

Once done replace train test and valid data files from here and it should work.

@kamalkraj @ranjeetds 
Hi,
I followed all  steps but I got an empty onto.train.ner, onto.developmnt.ner and onto.test.ner .
is there any ideas what should be the problem?
